x-pack/plugin/watcher/src/main/java/org/elasticsearch/xpack/watcher/trigger/schedule/engine/TickerScheduleTriggerEngine.java

```diff
@@ -106,13 +115,25 @@ public void add(Watch watch) {
 
     @Override
     public boolean remove(String jobId) {
+        logger.debug("Removing watch [{}] from engine (engine is running: {})", jobId, isRunning.get());
         return schedules.remove(jobId) != null;
     }
 
     void checkJobs() {
+        if (isRunning.get() == false) {
+            logger.debug(
+                "Watcher not running because the engine is paused. Currently scheduled watches being skipped: {}",
+                schedules.size()
+            );
+            return;
+        }
         long triggeredTime = clock.millis();
         List<TriggerEvent> events = new ArrayList<>();
         for (ActiveSchedule schedule : schedules.values()) {
+            if (isRunning.get() == false) {
```

Comment:
should this be moved a bit lower to the notify listeners and only notify them if it is running (else log the message) ? as-is there is not much difference between how race-like this can be. I think preventing the core action would help a bit more with the race like conditions.

Reply:
There were a couple of reasons I did it here instead:
(1) I was trying to modify as little code as possible to avoid introducing some new bug or making things worse
(2) The real problem (in my opinion) isn't if a single event happens to slip through after the `.watches` shard has been moved -- it's that we just keep running events until the end of time. This bails out pretty quickly once the `.watches` shard has been moved.